src/main/capi/arrow-c.cpp

```diff
@@ -108,3 +108,112 @@ duckdb_state duckdb_execute_prepared_arrow(duckdb_prepared_statement prepared_st
 	*out_result = (duckdb_arrow)arrow_wrapper;
 	return !arrow_wrapper->result->HasError() ? DuckDBSuccess : DuckDBError;
 }
+
+namespace arrow_array_stream_wrapper {
+namespace {
+struct PrivateData {
+	ArrowSchema *schema;
+	ArrowArray *array;
+	bool done;
+};
+
+void EmptySchemaRelease(ArrowSchema *schema) {
+}
+
+void GetSchemaVoid(uintptr_t stream_factory_ptr, duckdb::ArrowSchemaWrapper &schema) {
+	auto private_data = (PrivateData *)((ArrowArrayStream *)stream_factory_ptr)->private_data;
+	schema.arrow_schema = *private_data->schema;
+	schema.arrow_schema.release = EmptySchemaRelease;
+
+	for (int64_t i = 0; i < schema.arrow_schema.n_children; i++) {
+		schema.arrow_schema.children[i]->release = EmptySchemaRelease;
+	}
+}
+
+int GetSchema(struct ArrowArrayStream *stream, struct ArrowSchema *out) {
+	auto private_data = static_cast<arrow_array_stream_wrapper::PrivateData *>((stream->private_data));
+	*out = *private_data->schema;
+	return 0;
+}
+
+int GetNext(struct ArrowArrayStream *stream, struct ArrowArray *out) {
+	auto private_data = static_cast<arrow_array_stream_wrapper::PrivateData *>((stream->private_data));
+	*out = *private_data->array;
+	if (private_data->done) {
+		out->release(out);
+	}
+
+	private_data->done = true;
+	return 0;
+}
+
+const char *GetLastError(struct ArrowArrayStream *stream) {
+	return nullptr;
+}
```

Comment:
Shouldn't this return the stream last error?

Reply:
These functions are stubs that we use to wrap a given `ArrowArray` and `ArrowSchema` as an `ArrowStream`. From what I understand of Arrow, such a stream can never produce an error/i.e. there is no error to return because we already have the complete array and schema. Am I missing something? Please let me know if we want some comments to make this more apparent!

Reply:
I guess that makes sense. Thanks for the clarification!